lib/pages/image_picker_page.dart

```diff
@@ -0,0 +1,297 @@
+import 'dart:convert';
+
+import 'package:dotted_border/dotted_border.dart';
+import 'package:expandable/expandable.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:image_picker/image_picker.dart';
+import 'package:sentry_flutter/sentry_flutter.dart';
+import 'package:stable_horde_flutter/blocs/conversions_bloc.dart';
+import 'package:stable_horde_flutter/blocs/shared_prefs_bloc.dart';
+import 'package:stable_horde_flutter/dialogs/login_dialog.dart';
+import 'package:stable_horde_flutter/utils/image_picker_utils.dart';
+import 'package:stable_horde_flutter/widgets/section_frame.dart';
+
+class ImagePickerPage extends StatefulWidget {
+  const ImagePickerPage({super.key});
+
+  @override
+  State<ImagePickerPage> createState() => _ImagePickerPageState();
+}
+
+class _ImagePickerPageState extends State<ImagePickerPage> {
+  @override
+  Widget build(BuildContext context) {
+    return ExpandablePanel(
+      header: const Text(
+        "Img2Img",
+        style: TextStyle(fontSize: 18),
+      ),
+      collapsed: const SizedBox.shrink(),
+      expanded: _img2ImgOption(),
+      theme: const ExpandableThemeData(
+        iconColor: Colors.white,
+      ),
+    );
+  }
+
+  Widget _img2ImgOption() {
+    return FutureBuilder<String?>(
+      future: sharedPrefsBloc.getApiKey(),
+      builder: (context, snapshot) {
+        if (snapshot.hasError) {
+          print(snapshot.error);
+          print(snapshot.stackTrace);
+          Sentry.captureException(
+            snapshot.error,
+            stackTrace: snapshot.stackTrace,
+          );
+        }
+
+        if (snapshot.connectionState == ConnectionState.waiting) {
+          // Loading
+          return const Padding(
+            padding: EdgeInsets.all(12),
+            child: CircularProgressIndicator(),
+          );
+        }
+        var apiKey = snapshot.data;
+        if (apiKey == null) {
+          // User is anonymous
+          return _anonymouseWidget();
```

Comment:
I think we can simplify this code and UI a bit:
- Always show the "select your image" box regardless of whether or not the user is logged in.
- On tap, if the user is not logged in, show the dialog that says "You are currently anonymous. In order to use Img2Img login to a Stable Horde account."
- On tap, if the user is logged in, show the picker as normal.

This will reduce state management (it'll no longer be possible for the UI here to get out of sync with the login state)

Reply:
Sounds good, will modify the logic to follow above.